src/main/java/uk/gov/laa/pfla/auth/service/dao/MappingTableDao.java

```diff
@@ -1,53 +1,116 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
 import org.springframework.stereotype.Repository;
 import uk.gov.laa.pfla.auth.service.builders.MappingTableModelBuilder;
 import uk.gov.laa.pfla.auth.service.models.MappingTableModel;
 
+import java.sql.*;
 import java.util.ArrayList;
 import java.util.List;
+
 @Repository
-public class MappingTableDao {
+@Slf4j
+public class MappingTableDao  implements RowMapper<Object> {
 
     private final List<MappingTableModel> mappingTableObjectList = new ArrayList<>();
+
+//    @Autowired
+//    private JdbcTemplate jdbcTemplate;
+
+    @Value("${spring.datasource.url}")
+    private String databaseUrl;
+
+    @Value("${spring.datasource.username}")
+    private String databaseUsername;
+
+    @Value("${spring.datasource.password}")
+    private String databasePassword;
+
     public MappingTableDao() {
         //empty contructor to allow builder to do its work
     }
 
-    public List<MappingTableModel> fetchReportList() {
+    public Connection setupDB(){ //Todo - create custom exception
+
+        OracleDataSource ods = null;
+        try {
+                ods = new OracleDataSource();
+                ods.setURL(databaseUrl); // jdbc:oracle:thin@//[hostname]:[port]/[DB service name]
+                ods.setUser(databaseUsername);
+                ods.setPassword(databasePassword);
+            } catch(SQLException e){
+            log.error("Error in Oracle Datasource JDBC setup: " + e);
+        }
+
+        Connection conn = null;
+        try {
+             conn = ods.getConnection();
+        } catch(SQLException | NullPointerException e){
+            log.error("Error in creating DB Connection: " + e);
+        }
 
+
+        return conn;
+
+    }
+
+    public MappingTableModel mapRow(ResultSet resultSet, int rowNum) throws SQLException {
+
+        return new MappingTableModelBuilder()
+                .withId(resultSet.getInt(1))
+                .withReportName(resultSet.getString(2))
+                .withSqlString(resultSet.getString(3))
+                .withBaseUrl(resultSet.getString(4))
+                .withReportPeriod(resultSet.getString(5))
+                .withReportOwner(resultSet.getString(6))
+                .withReportCreator(resultSet.getString(7))
+                .withReportDescription(resultSet.getString(8))
+                .withExcelSheetNumber(resultSet.getInt(12))
+                .withCsvName(resultSet.getString(13))
+                .build();
+
+    }
+
+    public List<MappingTableModel> fetchReportList() {
+        ResultSet rslt;
+        int rowNumber = 0;
         mappingTableObjectList.clear(); // Prevent response data accumulating after multiple requests
+        MappingTableModel mappingTableObject;
+
 
-        //fetch data from database
-        //fetchSingleRowOfData()
+        Connection conn = null;
```

Comment:
Can we avoid calling this each time method is called? Jdbc template can be a better solution

Reply:
How does JDBC do it then - is it keeping a session open and reusing the same one till you close it or something?